pysmt/typing.py

```diff
@@ -65,7 +65,9 @@ def __eq__(self, other):
         return self.type_id == other.type_id
 
     def __ne__(self, other):
-        return not (self == other)
```

Comment:
Why this change? In general, this is not good, since we now require that any sub-class that modifies ```__eq__``` also  modifies ```__ne__```. See ```_BVType```.